airflow/api_internal/endpoints/rpc_api_endpoint.py

```diff
@@ -24,6 +24,7 @@
 
 from airflow.api_connexion.types import APIResponse
 from airflow.dag_processing.processor import DagFileProcessor
+from airflow.models.dag import DagModel
```

Comment:
Thinking again, it makes me uncomfortable that a module in `api_internal` is imported by something in `models`, but `models` is then imported into another module in `api_internal`. I understand this technically does not cause an import cycle, but it seems like an easy one to get wrong eventually by somebody.

Maybe it would be a good idea to put the imports here in a lazily-evaluated function? Or maybe we could build `METHODS_MAP` automatically when a function is decorated by `internal_api_call`?

Reply:
`internal_api_call` decorator is client-side, while `METHODS_MAP` is at server-side.

I agree that mixing the direction of imports may lead to the problems later.

What about simply wrapping the building of `METHODS_MAP` into a additional method (with imports there).
Like: https://github.com/apache/airflow/pull/28841/files

Reply:
I fixed the circular dependency by moving the `BaseSerialization` import. I think that should be enough?. I also updated the function `internal_api_call` to fix another issue. Feel free to take a look

Reply:
I still don’t like both `airflow.models` and `airflow.api_internal` import each other, although _technically_ the imports are not circular. Wrap `METHODS_MAP` in a function with local imports should be better.

Reply:
(We can do that in another PR though, this problem is common to most of the AIP-44 PRs.)